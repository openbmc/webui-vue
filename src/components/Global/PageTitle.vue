<template>
  <header class="page-title">
    <h1>{{ title }}</h1>
    <p v-if="description">{{ description }}</p>
  </header>
</template>

<script>
export default {
  name: 'PageTitle',
  props: ['description'],
  data() {
    return {
      title: this.$route.meta.title
    };
  }
};
</script>

<style lang="scss" scoped>
.page-title {
  margin-bottom: $spacer * 2;
}
h1 {
  font-weight: $display4-weight;
  line-height: $display-line-height;
}
p {
  max-width: 72ch;
}
</style>
