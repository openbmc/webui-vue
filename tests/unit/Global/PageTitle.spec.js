import { mount, createLocalVue } from '@vue/test-utils';
import PageTitle from '@/components/Global/PageTitle';

const localVue = createLocalVue();

describe('PageTitle.vue', () => {
  const wrapper = mount(PageTitle, {
    localVue,
    propsData: {
      description: 'A page title test description',
    },
    mocks: {
      $t: (key) => key,
      $route: {
        meta: {
          title: 'Page Title',
        },
      },
    },
  });
  it('should exist', () => {
    expect(wrapper.exists()).toBe(true);
  });
  it('should render h1 element', () => {
    expect(wrapper.find('h1').exists()).toBe(true);
  });
  it('should render p element', () => {
    expect(wrapper.find('p').exists()).toBe(true);
  });
  it('should render correctly', () => {
    expect(wrapper.element).toMatchSnapshot();
  });
});
