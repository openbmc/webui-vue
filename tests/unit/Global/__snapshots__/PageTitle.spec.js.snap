// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageTitle.vue should render correctly 1`] = `
<div
  class="page-title"
>
  <h1>
    Page Title
  </h1>
   
  <p>
    A page title test description
  </p>
</div>
`;
